plugins/soundsourcewv/audiosourcewv.cpp

```diff
@@ -0,0 +1,86 @@
+#include "audiosourcewv.h"
+
+#include "sampleutil.h"
+
+namespace Mixxx {
+
+AudioSourceWV::AudioSourceWV()
+        : m_wpc(NULL), m_sampleScale(0.0f) {
+}
+
+AudioSourceWV::~AudioSourceWV() {
+    close();
+}
+
+AudioSourcePointer AudioSourceWV::open(QString fileName) {
+    AudioSourceWV* pAudioSourceWV(new AudioSourceWV);
+    AudioSourcePointer pAudioSource(pAudioSourceWV); // take ownership
+    if (OK == pAudioSourceWV->postConstruct(fileName)) {
+        // success
+        return pAudioSource;
+    } else {
+        // failure
+        return AudioSourcePointer();
+    }
+}
+
+Result AudioSourceWV::postConstruct(QString fileName) {
+    char msg[80]; // hold possible error message
+    m_wpc = WavpackOpenFileInput(fileName.toLocal8Bit().constData(), msg,
+            OPEN_2CH_MAX | OPEN_WVC | OPEN_NORMALIZE, 0);
+    if (!m_wpc) {
+        qDebug() << "SSWV::open: failed to open file : " << msg;
+        return ERR;
+    }
+
+    setChannelCount(WavpackGetReducedChannels(m_wpc));
+    setFrameRate(WavpackGetSampleRate(m_wpc));
+    setFrameCount(WavpackGetNumSamples(m_wpc));
+
+    if (WavpackGetMode(m_wpc) & MODE_FLOAT) {
+        m_sampleScale = 1.0f;
+    } else {
+        const int bitsPerSample = WavpackGetBitsPerSample(m_wpc);
+        const uint32_t maxSampleValue = uint32_t(1) << (bitsPerSample - 1);
```

Comment:
Ah yes, sorry I must have have confused myself. 
Maybe it was because maxSampleValue should be 0x7FFF from the name. 
Can we call it sampleDivisor or something like that?